### ISSUE TITLE
`iterator` and `iteratorRecord` don't accept a Modifier

### ISSUE THREAD

**Issue by tstone** [outworkers/phantom#516]
It seems that `ExecutableQuery#iterator` and `ExecutableQuery#iteratorRecord` don't allow a Modifier to be specified in the same way that `fetch` and `fetchRecord` do.

Possible:  `mytable.select.fetch(_.setFetchSize(1024))`
Possible:  `mytable.select.fetchRecord(_.setFetchSize(1024))`

Not Possible:  `mytable.select.iterator(_.setFetchSize(1024))`
Not Possible:  `mytable.select.iteratorRecord(_.setFetchSize(1024))`

Is there a way to set the fetch size when using `iterator` or `iteratorRecord`?

**Comment by tstone** [comment]
It seems I figured out a way to achieve this with the present implementation:

```
val query mytable.select
query.future(_.setFetchSize(pageSize)).map {
  _.iterator().asScala.map(query.fromRow)
}
```

`ExecutableQuery#fromRow` is what I was missing.  This works but it would be nice™ if the interface for `iterator` and `iteratorRecord` matched the other methods.

One thing I'm wondering though is if `fromRow` is internal API and I shouldn't be using that?  It wasn't marked as `private` :)

**Comment by alexflav23** [comment]
Hi @tstone,

That's a very fair comment, it will make a nice addition and it will be inlcuded in the next release, as obviously it's quite easy to add the functionality. It's not marked as private such that people can work around issues just like you have, and inject the mapping ability just where it is needed.

Regards,
Flavian

**Comment by alexflav23** [comment]
Hi @tstone,

Fixed in 1.27.1, it will go live as part of the next release.

Regards.
